internal/controllers/provisioningrequest_hwprovision.go

```diff
@@ -132,11 +132,22 @@ func (t *provisioningRequestReconcilerTask) waitForHardwareData(
 	// Clear callback annotations after hardware processing to prevent false callback detection
 	// This ensures future reconciliations aren't incorrectly treated as callback-triggered
 	// If no annotations exist, this does nothing
-	ctlrutils.ClearPRCallbackAnnotations(t.object)
-	if err == nil {
-		t.logger.InfoContext(ctx, "Cleared PR callback annotations after successful hardware processing")
-	} else {
-		t.logger.InfoContext(ctx, "Cleared PR callback annotations after hardware processing completed with error")
+	annotations := t.object.GetAnnotations()
+	hasCallbackAnnotations := annotations != nil && annotations[ctlrutils.CallbackReceivedAnnotation] != ""
+
+	if hasCallbackAnnotations {
+		t.logger.InfoContext(ctx, "Clearing PR callback annotations after hardware processing",
+			"callbackReceived", annotations[ctlrutils.CallbackReceivedAnnotation],
+			"callbackStatus", annotations[ctlrutils.CallbackStatusAnnotation],
+			"narId", annotations[ctlrutils.CallbackNodeAllocationRequestIdAnnotation],
+			"hadError", err != nil)
+
+		// Clear annotations and persist to cluster to prevent false callback detection on next reconciliation
+		if clearErr := ctlrutils.ClearPRCallbackAnnotationsWithPatch(ctx, t.client, t.object); clearErr != nil {
+			t.logger.WarnContext(ctx, "Failed to clear PR callback annotations", "error", clearErr.Error())
+		} else {
+			t.logger.InfoContext(ctx, "Successfully cleared and persisted PR callback annotations")
```

Comment:
Isn't this part `and persisted PR callback annotations` implied in `Successfully cleared ...`?